### PR TITLE
bugfix/SOF-3813

### PR DESCRIPTION
**Pull request by mohammadidinani**

### CODE REVIEW

File: src/mixins/bonds.js [hunk 1]
```diff
@@ -34,7 +34,9 @@ export const BondsMixin = (superclass) => class extends superclass {
     areElementsBonded(element1, coordinate1, element2, coordinate2, bondsData) {
         const distance = Made.math.vDist(coordinate1, coordinate2);
         const connectivityFactor = this.settings.chemicalConnectivityFactor;
-        return Boolean(bondsData.find(b => b.length.value && (b.length.value * connectivityFactor) >= distance));
+        return Boolean(filterBondsDataByElementsAndOrder(bondsData, element1, element2).find(b => {
```
Comment (timurbazhirov) [inline src/mixins/bonds.js]: Let's clean up the docstrings in this file too, pls.

Comment (mohammadidinani) [reply src/mixins/bonds.js]: Done.

File: src/mixins/bonds.js [hunk 2]
```diff
@@ -123,7 +147,8 @@ export const BondsMixin = (superclass) => class extends superclass {
     }
 
     /**
-     * Iterates over all combination of atoms and draws bonds.
+     * Draw bonds. Bonds are created synchronously if the asynchronous callback (createBondsAsync) to draw bonds
+     * in background has not returned yet. This may happen if the structure is large and draw bonds is toggled quickly.
```
Comment (timurbazhirov) [inline src/mixins/bonds.js]: We need this to block the UI until the bonds are drawn.

Comment (mohammadidinani) [reply src/mixins/bonds.js]: Done.

**Pull request merged**